crates/primitives/src/bits/serde.rs

```diff
@@ -1,10 +1,31 @@
-use super::FixedBytes;
-use core::fmt;
+use super::{Address, FixedBytes};
+use alloc::string::String;
+use core::{fmt, str::FromStr};
 use serde::{
     de::{self, Visitor},
     Deserialize, Deserializer, Serialize, Serializer,
 };
 
+impl Serialize for Address {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let checksum_address = self.to_checksum(None);
+        serializer.serialize_str(&checksum_address)
```

Comment:
Use `to_checksum_buffer` (more efficient), and add the same `is_human_readable` as below in `impl<const N: usize> Serialize for FixedBytes<N>`